### PR TITLE
Add a button to copy/paste password reset urls

### PR DESCRIPTION
**Pull request by mia-pi-git**
Requested by aeonic.
Dumb hack, but there isn't really a better way to do this, is there?

### CODE REVIEW

File: website/users.php [hunk 1]
```diff
@@ -285,11 +285,30 @@
...
+			dummyLink.id = "dummyLink";
+			dummyLink.value = $("#token").text();
+			dummyLink.style.position = 'absolute';
+			appended.appendChild(dummyLink);
+			dummyLink.select();
+			document.execCommand("copy");
+			appended.removeChild(dummyLink);
+			$('#copy').text('Copied!');
+		}
+		</script>
```
Comment (Zarel) [inline website/users.php:303]: This doesn't work in searches. JS needs to go in `website/js/ladder.js`.

Comment (mia-pi-git) [reply website/users.php]: Done.

File: website/users.php [hunk 2]
```diff
@@ -289,7 +289,8 @@
 				Use this link:
 			</p>
 			<p style="margin: 1em -13px">
-				<small><code>https://<?= $psconfig['routes']['root'] ?>/resetpassword/<?php echo $token; ?></code></small>
+				<small><code id="token">https://<?= $psconfig['routes']['root'] ?>/resetpassword/<?php echo $token; ?></code></small>
```
Comment (Zarel) [inline website/users.php:292]: IDs are required to be unique by the spec; a static ID can't be unique if it's in a panel.

The best way to do this is to use a class and use a local search. You also want to replace the `onclick` handler with a Backbone click handler in `ladder.js`

Comment (Zarel) [reply website/users.php:292]: So find this:

```js
	events: {
		'change select[name=standing]': 'changeStanding',
		'click button[name=openReset]': 'openReset',
		'click button[name=cancelReset]': 'cancelReset'
	},
```

and add your copy button:

```js
	events: {
		'change select[name=standing]': 'changeStanding',
		'click button[name=copyLink]': 'copyLink',
		'click button[name=openReset]': 'openReset',
		'click button[name=cancelReset]': 'cancelReset'
	},
```

Comment (Zarel) [reply website/users.php:292]: Then you can follow along how `openReset` works.

Comment (mia-pi-git) [reply website/users.php]: Okay. Done.

File: website/js/ladder.js [hunk 3]
```diff
@@ -26,7 +26,23 @@ var LadderPanel = Panels.StaticPanel.extend({
 	events: {
 		'change select[name=standing]': 'changeStanding',
 		'click button[name=openReset]': 'openReset',
-		'click button[name=cancelReset]': 'cancelReset'
+		'click button[name=cancelReset]': 'cancelReset',
+		'click button[name=copyUrl]': 'copyUrl',
```
Comment (Zarel) [inline website/js/ladder.js:30]:
```suggestion
		'click button[name=copyUrl]': 'copyUrl'
```

We don't have eslint in `website/` but that's no reason to break ES3 compatibility!

Comment (mia-pi-git) [reply website/js/ladder.js:30]: this does what now
is it really _that_ frail?

Comment (mia-pi-git) [reply website/js/ladder.js]: smh ES3

File: website/js/ladder.js [hunk 4]
```diff
@@ -26,7 +26,23 @@ var LadderPanel = Panels.StaticPanel.extend({
...
+	},
+	copyUrl: function (e) {
+		var token = e.currentTarget.value;
+		// this works since this is in the root as well
+		var url = 'https://' + document.location.hostname + '/resetpassword/' + token;
+		var dummy = document.createElement("input");
+		var placeholder = $('.pfx-topbar')[0];
+		dummy.id = "dummy";
```
Comment (Zarel) [inline website/js/ladder.js:38]:
```suggestion
```
:|

While this ID is unlikely to cause harm, it's not impossible, and it's also pointless.

File: website/js/ladder.js [hunk 5]
```diff
@@ -26,7 +26,23 @@ var LadderPanel = Panels.StaticPanel.extend({
...
+		dummy.value = url;
+		dummy.style.position = 'absolute';
```
Comment (Zarel) [inline website/js/ladder.js:39]: does `display:none` not work here?

File: website/js/ladder.js [hunk 6]
```diff
@@ -26,7 +26,23 @@ var LadderPanel = Panels.StaticPanel.extend({
...
-		'click button[name=cancelReset]': 'cancelReset'
+		'click button[name=cancelReset]': 'cancelReset',
+		'click button[name=copyUrl]': 'copyUrl',
+	},
+	copyUrl: function (e) {
+		var token = e.currentTarget.value;
+		// this works since this is in the root as well
+		var url = 'https://' + document.location.hostname + '/resetpassword/' + token;
+		var dummy = document.createElement("input");
+		var placeholder = $('.pfx-topbar')[0];
```
Comment (Zarel) [inline website/js/ladder.js:37]: Why not just `document.body`?

File: website/js/ladder.js (hunk 5 above)
Comment (mia-pi-git) [reply website/js/ladder.js:39]: Probably, i just stole this from the other place we do copying that's not in the main site.

File: website/js/ladder.js [hunk 7]
```diff
@@ -67,7 +81,7 @@ var App = Panels.App.extend({
 		'usersearch/': LadderPanel,
 		'usersearch/:q': LadderPanel,
 		'users/:user/modlog': LadderPanel
-	}
+	},
```
Comment (Zarel) [inline website/js/ladder.js:84]:
```suggestion
	}
```

...ES3!

**Comment by Zarel** [comment]
Should be good now. Thanks!

**Comment by Zarel** [comment]
Still surprised people care, though. Triple-click + Cmd+C is what I do. You need your keyboard for Cmd+V anyway.

**Pull request merged**